Backend/TimeTracker.Api/Controllers/AuthController.cs

```diff
@@ -135,9 +136,23 @@ public async Task<GenericResponseDTO<int>> Register(UserDTO registerData)
                     CreatedTime = DateTime.UtcNow,
                     Email = registerData.Email,
                     Password = authHelper.GetPasswordHash(registerData.Password, configuration),
-                    Name = registerData.Name
+                    Name = registerData.Name,
+                    Projects = new List<Project>()
                 };
 
+                // check if the user registered with an invite code, if they did, add them to a project
+                if(registerData.InviteCode != null && !registerData.InviteCode.SequenceEqual("")) {
+
+                    Project project = await database.Projects
+                        .FirstOrDefaultAsync(p => p.InviteCode == registerData.InviteCode);
+
+                    if(project != null) {
+                        newUser.Projects.Add(project);
+                        project.Students.Add(newUser);
```

Comment:
These are basically linked to the database so you really only need 1 of these lines. Either will work just fine.